osf_tests/test_analytics.py

```diff
@@ -42,3 +44,125 @@ def test_increment_user_activity_counters(self):
         assert_equal(user.get_activity_points(), 0)
         analytics.increment_user_activity_counters(user._id, 'project_created', date.isoformat())
         assert_equal(user.get_activity_points(), 1)
+
+
+@pytest.fixture()
+def user():
+    return UserFactory()
+
+
+@pytest.fixture()
+def project():
+    return ProjectFactory()
+
+@pytest.fixture()
+def project_with_contrib(user):
+    return ProjectFactory(creator=user)
+
+
+@pytest.fixture()
+def file_node(project):
+    file_node = OsfStorageFile(name='test', node=project)
+    file_node.save()
+    return file_node
+
+@pytest.fixture()
+def file_node2(project):
+    file_node2 = OsfStorageFile(name='test2', node=project)
+    file_node2.save()
+    return file_node2
+
+@pytest.fixture()
+def file_node3(project):
+    file_node2 = OsfStorageFile(name='test3', node=project)
+    file_node2.save()
+    return file_node2
+
+@pytest.fixture()
+def page_counter(project, file_node):
+    page_counter_id = 'download:{}:{}'.format(project._id, file_node.id)
+    page_counter, created = PageCounter.objects.select_for_update().get_or_create(_id=page_counter_id)
+    page_counter.date = {u'2018/02/04': {u'total': 41, u'unique': 33}}
+    page_counter.save()
+    return page_counter
+
+@pytest.fixture()
+def page_counter2(project, file_node2):
```

Comment:
Are these duplicative fixtures necessary? Can we test the behavior with just one fixture for each type of object?

Reply:
Yeah this does look confusing, but I want to show that we are summing over multiple PageCounter objects and each one corresponds with a file node. I'll try to clean it up and add comments. I'm not sure if this is a good test if there's isn't summing across multiple PageCounter objects.